mission_control/navigator_alarm/navigator_alarm_handlers/thruster_fault.py

```diff
@@ -0,0 +1,49 @@
+from ros_alarms import HandlerBase, AlarmBroadcaster
+from roboteq_msgs.msg import Status
+import rospy
+
+
+class ThrusterFault(HandlerBase):
+    alarm_name = 'thruster-fault'
+
+    def __init__(self):
+        self.broadcaster = AlarmBroadcaster(self.alarm_name)
+
+        # Thruster topics to listen to
+        motor_topics = ['/FL_motor', '/FR_motor', '/BL_motor', '/BR_motor']
+
+        # Subscribe to the status message for all thruster topics
+        status_sub = [rospy.Subscriber(
+            topic + '/status', Status, self._check_faults, topic) for topic in motor_topics]
+
+        # Dictionary for the faults as defined in roboteq_msgs/Status
+        self.fault_codes = {1: 'OVERHEAT', 2: 'OVERVOLTAGE', 4: 'UNDERVOLTAGE', 8: 'SHORT_CIRCUIT',
+                            16: 'EMERGENCY_STOP', 32: 'SEPEX_EXCITATION_FAULT', 64: 'MOSFET_FAILURE', 128: 'STARTUP_CONFIG_FAULT'}
+
+        self._raised = False
+
+    def _check_faults(self, msg, topic):
+        if msg.fault == 0:
+            return
+
+        # Figure out what fault codes have been raised and find their meaning from dictionary
+        get_fault_codes = []
+        for i in range(0, len(self.fault_codes)):
+            # Fault message is the sum of binary strings
+            decode = msg.fault & 2**i
```

Comment:
Shouldn't this just be
```
for key in self.fault_codes:
    decode = msg.fault & key
```
I don't think you need to raise 2 to i. 2^128 is way beyond integer limit

Reply:
I guess that's the same thing cause 2^i gives 0,1,2,...,128. Using key is a little more efficient i suppose